commit mongo code, user data and routes, server

GraceMattern opened the pull request.

maorm200 commented on data/users.js:
```diff
@@ -0,0 +1,168 @@
+    const userList = await userCollection.find({}).toArray();
+    
+    if(!userList) throw `No users in the system`
+    for (let i = 0; i < userList.length; i++) {
```
Maybe you can simplify the for loop to a for(let x of userList) type of loop

GraceMattern replied:
Implemented
0. changed for loop to for in loop
1. refined isEmail to include banks that are longer than one word
2. removed age restrictions in isAdult since account collection will lookup age and then allow certain account types 
3. added userId to user schema

GraceMattern replied:
Implemented

maorm200 reviewed (changes requested):
Added just one comment on code, rest looks fine!

GraceMattern commented:
1. refined isEmail to include banks that are longer than one word
2. removed age restrictions in isAdult since account collection will lookup age and then allow certain account types 
3. added userId to user schema

maorm200 reviewed (changes requested):
```
for (let i in userList) {
        userList[i]._id = userList[i]._id.toString(); 
    }
```

Just be careful as you still need to change inside the for loop you no longer need [i] as you no longer are using the former loop.

GraceMattern commented:
> ```
> for (let i in userList) {
>         userList[i]._id = userList[i]._id.toString(); 
>     }
> ```
> 
> Just be careful as you still need to change inside the for loop you no longer need [i] as you no longer are using the former loop.

like this?
```
for (let i of userList) {
        i._id = i._id.toString(); 
```

maorm200 commented:
Yes, that is correct!
